### PR TITLE
Minileven: add options back  as they still exist on sites

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

This is a follow-up to #14714.
After removing those options, we started seeing errors like this one on sites:
PHP Warning: Invalid Jetpack option name: wp_mobile_excerpt

Since these Jetpack options are still valid options even if the feature is gone, let's keep them around.

#### Testing instructions:

* Start from a site running Jetpack stable.
* with the Beta plugin, switch from stable to this branch.
* You should see no notices.

#### Proposed changelog entry for your changes:

* N/A

### CODE REVIEW

**Comment by roccotripaldi** [comment]
Following your testing instructions, I did get a bunch of notices:
```

Warning: Invalid Jetpack option name: wp_mobile_excerpt in /srv/users/user116b676c/apps/user116b676c/public/wp-content/plugins/jetpack-dev/vendor/automattic/jetpack-options/legacy/class-jetpack-options.php on line 189

Warning: Invalid Jetpack option name: wp_mobile_featured_images in /srv/users/user116b676c/apps/user116b676c/public/wp-content/plugins/jetpack-dev/vendor/automattic/jetpack-options/legacy/class-jetpack-options.php on line 189

Warning: Invalid Jetpack option name: wp_mobile_app_promos in /srv/users/user116b676c/apps/user116b676c/public/wp-content/plugins/jetpack-dev/vendor/automattic/jetpack-options/legacy/class-jetpack-options.php on line 189

```

**Comment by kbrown9** [comment]
It looks like the problem is that the three `wp_mobile_*` options that we delete aren’t managed by Jetpack_Options. So, Jetpack_Options::get_option() doesn’t recognize them as valid option names.

These options were originally in the array returned by [Jetpack_Options::get_all_wp_options()](https://github.com/Automattic/jetpack/pull/14714/files#diff-f6704854b359bef524e3364944794b0cL561) The options in that array aren’t managed by Jetpack_Options.

I think we need to call core’s get_option() and delete_option() functions to get and delete these options in Jetpack::plugin_upgrade().

**Comment by jeherve** [comment]
Thank you. That makes sense now, I was confused. I updated the calls, this should be ready for another review now.

File: packages/options/legacy/class-jetpack-options.php [hunk 1]
```diff
@@ -558,6 +558,9 @@ public static function get_all_wp_options() {
 			'disabled_likes',
 			'disabled_reblogs',
 			'jetpack_comments_likes_enabled',
+			'wp_mobile_excerpt',
```
Comment (kraftbj) [inline packages/options/legacy/class-jetpack-options.php]: I think we can still get rid of these. This function is called when JP is deleted or having the options reset, but they would already have been deleted in `plugin_upgrade`

**Review by kraftbj — DISMISSED** [review]

**Comment by kraftbj** [comment]
As discussed in Slack, we need to take into account the spun-out plugin so need to gate this to only delete options if the plugin is not active.

File: packages/options/legacy/class-jetpack-options.php (hunk 1 above)
Comment (jeherve) [reply packages/options/legacy/class-jetpack-options.php]: Done in eb9b8991b83569107bb63538c1f3c9092268bd4c

**Comment by jeherve** [comment]
> we need to take into account the spun-out plugin so need to gate this to only delete options if the plugin is not active.

I gave this a try in 0060167eaea76e9694699ac1c177262e2d1bc65f. Let me know if this is what you had in mind!

**Review by kbrown9 — APPROVED** [review]
This looks good to me! The PHP warnings are no longer generated.

**Pull request merged**